BridgeCareApp/BridgeCareCodeFirst/Models/DeficientData.cs

```diff
@@ -0,0 +1,125 @@
+{
+    public class DeficientData
+    {
+        public DeficientOrTargetTab GetDeficient(SimulationResult data, int[] totalYears)
+        {
+            var db = new BridgeCareContext();
+            IQueryable<DeficientResult> deficientOrTargetList = null;
+            DeficientOrTargetTab Results = null;
+
+            string getReport =
+                "SELECT TargetID, Years, TargetMet, IsDeficient " +
+                    " FROM Target_" + data.NetworkId
+                    + "_" + data.SimulationId;
+
+            try
+            {
+                var rawDeficientList = db.Database.SqlQuery<DeficientResult>(getReport).AsQueryable();
+
+                deficientOrTargetList = rawDeficientList.Where(_ => _.IsDeficient == true);
+
+                var dataList = new DeficientOrTarget();
+                var targetAndYear = dataList.DeficientTargetList(deficientOrTargetList);
+                var deficient = new DeficientData();
+                Results = deficient.DeficientInfo(data, targetAndYear, totalYears);
+            }
+            catch (SqlException ex)
+            {
+                db.Dispose();
+                if (ex.Number == -2 || ex.Number == 11)
+                {
+                    throw new TimeoutException("The server has timed out. Please try after some time");
+                }
+                if (ex.Number == 208)
+                {
+                    throw new InvalidOperationException("Target table does not exist in the database");
+                }
+            }
+            catch (OutOfMemoryException)
+            {
+                db.Dispose();
+                throw new OutOfMemoryException("The server is out of memory. Please try after some time");
+            }
+            catch (Exception ex)
+            {
+                db.Dispose();
+                Console.WriteLine(ex);
+            }
+            db.Dispose();
+            return Results;
+        }
+
+        public DeficientOrTargetTab DeficientInfo(SimulationResult data, Hashtable YearsIDValues, int[] totalYears)
+        {
+            BridgeCareContext db = new BridgeCareContext();
+            var deficientTableData = db.Deficient.AsNoTracking().Where(_ => _.SimulationID == data.SimulationId);
+            var totalYearCount = totalYears.Count();
+            DataTable DeficientTable = new DataTable();
+            DeficientTable.Columns.Add("Attribute");
+            DeficientTable.Columns.Add("Group");
+            Dictionary<int, List<int>> deficientList = new Dictionary<int, List<int>>();
+            for (int i = 0; i < totalYearCount; i++)
+            {
+                DeficientTable.Columns.Add(totalYears[i].ToString());
+            }
+            var increment = 0;
+            foreach (var item in deficientTableData)
+            {
+                Hashtable yearValues = new Hashtable();
+                yearValues = (Hashtable)YearsIDValues[item.Id_];
+
+                DataRow newDataRow = DeficientTable.NewRow();
```

Comment:
Consider reviewing the reports from legacy-iAM to see how the information was loaded into Excel. Discuss with William.